hackstar.py

```diff
@@ -336,6 +336,82 @@ def stats():
     return render_template("stats.html", **stats_data)
 
 
+@app.route("/songs")
+def songs():
+    db = get_db()
+    cursor = db.cursor()
+    
+    # Get all songs with their job status using JOIN
+    songs_query = """
+        SELECT s.id, s.title, s.artist, s.release_date, s.cover, 
+               f.state as file_state, f.filename, 
+               u.state as url_state, u.url
+        FROM song s
+        LEFT OUTER JOIN job_url u ON s.id = u.song_id
+        LEFT OUTER JOIN job_file f ON s.id = f.song_id
+        ORDER BY s.id DESC
+    """
+    songs = cursor.execute(songs_query).fetchall()
+    
+    # Process the results
+    songs_with_status = []
+    for song in songs:
+        song_id, title, artist, release_date, cover, file_state, filename, url_state, url = song
+        
+        # Determine import status and source
+        import_status = "unknown"
+        import_source = "unknown"
+        
+        if url_state:
+            import_status = url_state
+            import_source = f"URL: {url[:50]}..." if url and len(url) > 50 else f"URL: {url}" if url else "URL: unknown"
+        elif file_state:
+            import_status = file_state
+            import_source = f"File: {filename}" if filename else "File: unknown"
```

Comment:
This can be:
```suggestion
        import_status = url_state or file_state
        import_source = url or filename
```
Also, if the URL is too long, we can/should use CSS (overflow) to hide the remainder of the URL and add an ellipsis.